MapboxDirections/MBRouteOptions.swift

```diff
@@ -0,0 +1,276 @@
+     */
+    public convenience init(locations: [CLLocation], profileIdentifier: String? = nil) {
+        let waypoints = locations.map { Waypoint(location: $0) }
+        self.init(waypoints: waypoints, profileIdentifier: profileIdentifier)
+    }
+    
+    /**
+     Initializes a route options object for routes between the given geographic coordinates and an optional profile identifier.
+     
+     - parameter coordinates: An array of geographic coordinates representing locations that the route should visit in chronological order. The array should contain at least two locations (the source and destination) and at most 25 locations. Each coordinate is converted into a `Waypoint` object.
+     - parameter profileIdentifier: A string specifying the primary mode of transportation for the routes. This parameter, if set, should be set to `MBDirectionsProfileIdentifierAutomobile`, `MBDirectionsProfileIdentifierCycling`, or `MBDirectionsProfileIdentifierWalking`. `MBDirectionsProfileIdentifierAutomobile` is used by default.
+     */
+    public convenience init(coordinates: [CLLocationCoordinate2D], profileIdentifier: String? = nil) {
+        let waypoints = coordinates.map { Waypoint(coordinate: $0) }
+        self.init(waypoints: waypoints, profileIdentifier: profileIdentifier)
+    }
+    
+    // MARK: Specifying the Path of the Route
+    
+    /**
+     An array of `Waypoint` objects representing locations that the route should visit in chronological order.
+     
+     A waypoint object indicates a location to visit, as well as an optional heading from which to approach the location. 
+     
+     The array should contain at least two waypoints (the source and destination) and at most 25 waypoints.
+     */
+    public var waypoints: [Waypoint]
+    
+    /**
+     A Boolean value that indicates whether a returned route may require a point U-turn at an intermediate waypoint.
+     
+     If the value of this property is `true`, a returned route may require an immediate U-turn at an intermediate waypoint. At an intermediate waypoint, if the value of this property is `false`, each returned route may continue straight ahead or turn to either side but may not U-turn. This property has no effect if only two waypoints are specified.
+     
+     Set this property to `true` if you expect the user to traverse each leg of the trip separately. For example, it would be quite easy for the user to effectively “U-turn” at a waypoint if the user first parks the car and patronizes a restaurant there before embarking on the next leg of the trip. Set this property to `false` if you expect the user to proceed to the next waypoint immediately upon arrival. For example, if the user only needs to drop off a passenger or package at the waypoint before continuing, it would be inconvenient to perform a U-turn at that location.
```

Comment:
Here’s the documentation corresponding to the `continue_straight` query parameter.